src/app/app.component.ts

```diff
@@ -174,17 +174,34 @@ export class AppComponent implements OnInit, OnDestroy {
     const routesWithoutParent = [];
     extensionRoutes.forEach((extensionRoute: ExtensionRoute) => {
       if (this.extensionRouteHasChild(extensionRoute)) {
-        const routeIndex = this.router.config.findIndex(
-          route => route.path === extensionRoute.parentRoute
-        );
-        this.convertExtensionRouteToRoute(extensionRoute);
-        this.router.config[routeIndex].children.unshift(extensionRoute);
+        // const routeIndex = this.router.config.findIndex(
```

Comment:
should probably be removed?

Reply:
I would like to move the whole logic into the app.extension.service.ts, there is no component related logic here. This PR is half-baked yet (draft), please don't be bothered with it, I just pushed my changes before switching to regression and blocker issues. :)